experimental/skills/DialogToDialog/DialogRootBot/Dialogs/SkillDialog.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Integration.AspNet.Core.Skills;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.BotBuilderSamples.DialogRootBot.Extensions;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.BotBuilderSamples.DialogRootBot.Dialogs
+{
+    /// <summary>
+    /// A sample dialog that can wrap remote calls to a skill.
+    /// </summary>
+    /// <remarks>
+    /// The options parameter in <see cref="BeginDialogAsync"/> must be a <see cref="SkillDialogArgs"/> instance
+    /// with the initial parameters for the dialog.
+    /// </remarks>
+    public class SkillDialog : Dialog
+    {
+        private readonly IStatePropertyAccessor<string> _activeSkillProperty;
+        private readonly string _botId;
+        private readonly ConversationState _conversationState;
+        private readonly SkillHttpClient _skillClient;
+        private readonly SkillsConfiguration _skillsConfig;
+
+        public SkillDialog(ConversationState conversationState, SkillHttpClient skillClient, SkillsConfiguration skillsConfig, IConfiguration configuration)
+            : base(nameof(SkillDialog))
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            _botId = configuration.GetSection(MicrosoftAppCredentials.MicrosoftAppIdKey)?.Value;
+            if (string.IsNullOrWhiteSpace(_botId))
+            {
+                throw new ArgumentException($"{MicrosoftAppCredentials.MicrosoftAppIdKey} is not in configuration");
+            }
+
+            _skillClient = skillClient ?? throw new ArgumentNullException(nameof(skillClient));
+            _skillsConfig = skillsConfig ?? throw new ArgumentNullException(nameof(skillsConfig));
+            _conversationState = conversationState ?? throw new ArgumentNullException(nameof(conversationState));
+            _activeSkillProperty = conversationState.CreateProperty<string>($"{typeof(SkillDialog).FullName}.ActiveSkillProperty");
+        }
+
+        public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default)
+        {
+            if (options is CancellationToken)
```

Comment:
Is this test necessary?

Reply:
Not really, removed it, thanks.

---
In reply to: [356227879](https://github.com/microsoft/BotBuilder-Samples/pull/2032#discussion_r356227879) [](ancestors = 356227879)